iroh/src/client/blobs/batch.rs

```diff
@@ -0,0 +1,460 @@
+        // will be cleaned up as soon as the entire batch is dropped.
+        //
+        // E.g. a typical scenario is that you create a large array of temp tags, and then
+        // store them in a hash sequence and then drop the array. You will get many drops
+        // at the same time, and might get a send failure here.
+        //
+        // But that just means that the server will clean up the temp tags when the batch is
+        // dropped.
+        updates.feed(BatchUpdate::Drop(*content)).now_or_never();
+        updates.flush().now_or_never();
+    }
+}
+
+/// Options for adding a file as a blob
+#[derive(Debug, Clone, Copy, Default)]
+pub struct AddFileOpts {
+    /// The import mode
+    pub import_mode: ImportMode,
+    /// The format of the blob
+    pub format: BlobFormat,
+}
+
+/// Options for adding a directory as a collection
+#[derive(Debug, Clone)]
+pub struct AddDirOpts {
+    /// The import mode
+    pub import_mode: ImportMode,
+    /// Whether to preserve the directory name
+    pub wrap: WrapOption,
+    /// Io parallelism
+    pub io_parallelism: usize,
+}
+
+impl Default for AddDirOpts {
+    fn default() -> Self {
+        Self {
+            import_mode: ImportMode::TryReference,
+            wrap: WrapOption::NoWrap,
+            io_parallelism: 4,
+        }
+    }
+}
+
+/// Options for adding a directory as a collection
+#[derive(Debug, Clone)]
+pub struct AddReaderOpts {
+    /// The format of the blob
+    pub format: BlobFormat,
+    /// Size of the chunks to send
+    pub chunk_size: usize,
+}
+
+impl Default for AddReaderOpts {
+    fn default() -> Self {
+        Self {
+            format: BlobFormat::Raw,
+            chunk_size: 1024 * 64,
+        }
+    }
+}
+
+impl Batch {
+    pub(super) fn new(
+        batch: BatchId,
+        rpc: RpcClient,
+        updates: UpdateSink<RpcService, RpcConnection, BatchUpdate>,
+        buffer_size: usize,
+    ) -> Self {
+        let updates = updates.buffer(buffer_size);
+        Self(Arc::new(BatchInner {
+            batch,
+            rpc,
+            updates: updates.into(),
+        }))
+    }
+
+    /// Write a blob by passing bytes.
+    pub async fn add_bytes(&self, bytes: impl Into<Bytes>) -> Result<TempTag> {
+        self.add_bytes_with_opts(bytes, Default::default()).await
+    }
+
+    /// Import a blob from a filesystem path, using the default options.
+    ///
+    /// For more control, use [`Self::add_file_with_opts`].
+    pub async fn add_file(&self, path: PathBuf) -> Result<(TempTag, u64)> {
+        self.add_file_with_opts(path, AddFileOpts::default()).await
+    }
+
+    /// Add a directory as a hashseq in iroh collection format
+    pub async fn add_dir(&self, root: PathBuf) -> Result<TempTag> {
+        self.add_dir_with_opts(root, Default::default()).await
+    }
+
+    /// Write a blob by passing an async reader.
+    ///
+    /// This will use a default chunk size of 64KB, and a format of [BlobFormat::Raw].
```

Comment:
```suggestion
    /// This will consume the stream in 64KB chunks, and use a format of [BlobFormat::Raw].
    ///
    /// For more options, see [`add_dir_with_opts`].
```